t/43-cli-api.t

```diff
@@ -495,4 +495,36 @@ subtest 'PIPE input' => sub {
     is $data->{body}, 'Hello openQA!', 'request body';
 };
 
+subtest 'YAML Templates' => sub {
+    my ($stdout, $stderr, @result) = capture sub { $api->run(@host, '-X', 'POST', 'job_groups', 'name=Test'); };
+    my $data = decode_json $stdout;
+    is $data->{id}, '1', 'create job group';
+    is_deeply \@result, [0], 'zero exit code';
+    is $stderr, '', 'quiet';
+    my $yaml_text = "products: {}\nscenarios: {}";
+    ($stdout, $stderr, @result) = capture sub {
+        $api->run(@host, '-X', 'POST', 'job_templates_scheduling/1', 'schema=JobTemplates-01.yaml', 'preview=1',
+            "template=$yaml_text");
+    };
+    $data = decode_json $stdout;
+    is $data->{job_group_id}, '1', 'YAML Text';
+    is_deeply \@result, [0], 'zero exit code';
```

Comment:
I already approved the PR but in case you still have a bit of time to improve or revisit this anyway then I suggest to improve the test strings with a bit more detail and also make them unique, e.g. 'zero exit code' is used multiple times. It's easier to pinpoint the corresponding source code from failing tests if the test messages are unambiguous

Reply:
Sure. Will do that. Used mostly the same names as it seemed to be some sort of standard, but I'm happy to change them.